### PR TITLE
refactor: Make logging more structured and consistent

### PR DESCRIPTION
**Pull request by techeclecticdesign**
This is a WIP.  I think it could be better, and some feedback would be welcome.
In particular I am wavering on logging errors where ORM calls fail.  Either way the error message generated by GORM would be propagated to the handlers.  I'm not sure if I should log every failure in backend/databases/, or cut back on the ones that are there now.  My initial thought is instead of logging a separate entry, instead returning a modified error object that includes both the GORM error message and some information about which action failed, which then gets presented when an error is thrown in the handler.
I'm also interested in hearing in whether I should include more fields in the handlers.  I went pretty lightweight in terms of what data I included, which is both readable and scannable.  There is a lot of potentially useful information in the request headers, but I don't want to make it too spammy, either.

### CODE REVIEW

File: backend/src/handlers/actions.go [hunk 1]
```diff
@@ -60,19 +68,25 @@ func (srv *Server) HandleImportUsers(w http.ResponseWriter, r *http.Request) {
 }
 
 func (srv *Server) HandleImportPrograms(w http.ResponseWriter, r *http.Request) {
+	logFields := log.Fields{
```
Comment (PThorpe92) [inline backend/src/handlers/actions.go:71]: I like this method of declaring the log fields, and making sure the handler name is one of them each time :+1:

**Comment by PThorpe92** [comment]
All around looks really good I'm super happy with it :+1: 

I know what you mean as far as the database calls. I recently just started returning the error instead of logging and propogating up
```go
// e.g. instead of
 if err := Db.Conn.Table("users").Where("id = ?", userId).Find(&users).Error; err != nil {
    log.Error("error retrieving users")
   return nil, err
} 

// have been switching to this:
return users, Db.Conn.Table("users").Where("id = ?", userId).Find(&users).Error
}
```
Which will cut down on that.


This is super helpful, we will be in a good place with logging. Much appreciated :+1:

**Comment by techeclecticdesign** [comment]
I redid the database portion, and added a helper function, LogDbError, to reduce boilerplate and standardize error logging on the DB side.  It does propagate calls, but it makes the error messages on the handler side include more context than the very vague GORM errors.  On the database side the errors include context such as what method the error was in, along with an error message describing the query that failed.  Let me know if you like this, or if I maybe overthought things.  Error handling is an important and often overlooked part of a program, and I am happy to put time in to make this as good as possible.

**Comment by techeclecticdesign** [comment]
I only just saw your comment earlier, and reflecting on it, I am going to make one quick set of changes so that there is only one error reported when database has an error.  Be done this weekend.

**Comment by techeclecticdesign** [comment]
Couple last minute changes that make this a little better:
I eliminated 2nd argument of LogDbError because half the time it was repeating itself from the error message in handler.
I added in the method with the failing GORM call as part of the error message formed by LogDbError.

**Review by PThorpe92 — APPROVED** [review]
This took a while to fix all the merge conflicts because of how many changes have been made in the past couple weeks.
Gonna see if the tests all pass, but I was pretty careful about it, and I might still pull it down and test again locally.
But thanks for the PR, this greatly improves our logging situation :+1:

**Pull request merged**